demos/http/http_demo_s3_download/CMakeLists.txt

```diff
@@ -0,0 +1,39 @@
+set( DEMO_NAME "http_demo_s3_download" )
+# Demo target.
+add_executable(${DEMO_NAME})
+
+set(OPENSSL_USE_STATIC_LIBS TRUE)
+find_package(OpenSSL REQUIRED)
+
+set( CMAKE_THREAD_PREFER_PTHREAD ON )
+find_package(Threads REQUIRED)
+
+target_sources(
+    ${DEMO_NAME}
+    PRIVATE
+        "${DEMO_NAME}.c"
+)
+
+target_link_libraries(
+    ${DEMO_NAME}
+    PRIVATE
+        http
+        OpenSSL::Crypto
+        OpenSSL::SSL
+	# SSL uses Threads and on some platforms require explicit linking.
+        Threads::Threads
+	# SSL uses Dynamic Loading and on some platforms requires explicit linking.
+        ${CMAKE_DL_LIBS}
+)
+
+target_include_directories(
+    http
+    PUBLIC
+        ${CMAKE_CURRENT_LIST_DIR}
+)
+
+target_include_directories(
+    ${DEMO_NAME}
+    PUBLIC
+        ${LOGGING_INCLUDE_DIRS}
+)
```

Comment:
Let's bring the python script and README from AFR to run the S3 demo. There will be changes for this repo